flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/ExecutionGraph.java

```diff
@@ -512,24 +513,19 @@ public boolean isArchived() {
 	}
 
 	public void enableCheckpointing(
-			long interval,
-			long checkpointTimeout,
-			long minPauseBetweenCheckpoints,
-			int maxConcurrentCheckpoints,
-			CheckpointRetentionPolicy retentionPolicy,
+			CheckpointCoordinatorConfiguration chkConfig,
 			List<ExecutionJobVertex> verticesToTrigger,
 			List<ExecutionJobVertex> verticesToWaitFor,
 			List<ExecutionJobVertex> verticesToCommitTo,
 			List<MasterTriggerRestoreHook<?>> masterHooks,
 			CheckpointIDCounter checkpointIDCounter,
 			CompletedCheckpointStore checkpointStore,
 			StateBackend checkpointStateBackend,
-			CheckpointStatsTracker statsTracker,
-			boolean isPreferCheckpointForRecovery) {
+			CheckpointStatsTracker statsTracker) {
 
 		// simple sanity checks
-		checkArgument(interval >= 10, "checkpoint interval must not be below 10ms");
-		checkArgument(checkpointTimeout >= 10, "checkpoint timeout must not be below 10ms");
+		checkArgument(chkConfig.getCheckpointInterval() >= 10, "checkpoint interval must not be below 10ms");
```

Comment:
As we have `CheckpointCoordinatorConfiguration` could we just place all the checks in the constructor of `CheckpointCoordinatorConfiguration`?

Reply:
Actually, the interval in many places is not united. In CheckpointConfig it used -1 as a disabled flag and the value which large than 0 is legal. So in CheckpointCoordinatorConfiguration the sanity check based on 1. It's better to unify them. Either -1 or larger than or equal to 10. So agree with you!